nomenclature/codelist.py

```diff
@@ -562,14 +562,14 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
             for top_level_cat in _code_list:
                 for top_key, _codes in top_level_cat.items():
                     for item in _codes:
-                        code = Code.from_dict(item)
+                        code = RegionCode.from_dict(item)
```

Comment:
You could streamline this slightly: 
```suggestion
                        code_list.append(RegionCode.from_dict({
                        **item,
                         "hierarchy": top_key,
                          "file":yaml_file.relative_to(path.parent).as_posix()})
                          )
```
although I'm not sure it's much more readable.

Reply:
As discussed bilaterally, please disregard.